fastlane/lib/fastlane/actions/notarize.rb

```diff
@@ -44,7 +45,7 @@ def self.run(params)
           log: verbose
         )
 
-        FileUtils.rm_rf(compressed_package_path) if compressed_package_path
+        FileUtils.rm_rf(compressed_package_path) if compressed_package_path && remove_compressed_package
```

Comment:
Question! Instead of introduction a new package `remove_compressed_package `, could we change this to `package_path != compressed_package_path `?

This why the user doesn’t need to set any options or anything 😇 Thoughts?

Reply:
But `compressed_package_path` is `package_path`+`.zip`. So this comparison would always be true.

Reply:
We could add a parameter to specify the `compressed_package_path` and then do something like that?

Reply:
Ohhh, one more question which might help me out 🙃 What is the purpose of uploading the `.zip` file? After notarization, a ticket gets stapled to the `.app` so I _think_ you don’t even need the `.zip`... unless there is something I’m missing which is totally possible 😛

Reply:
Sure! So notarization is for distributing apps outside of the AppStore, and you can choose to distribute the app via zip, if that's the case, the zip that was notarized is the one that needs to be distributed. Otherwise it won't run.
I use app center for distribution, and the appcenter plugin needs a zip.
I invoke it like this after notarization:
```fastfile
appcenter_upload(
      api_token: "aaabbbcccdddeeefff",
      owner_name: "username",
      owner_type: "user",
      app_name: "AppName",
      file: "artifacts/AppName.app.zip"
    )
```

Reply:
Ah, weird! I’ve notarized a macOS app before and the ticket was getting stapled to my `.app`.  You can see below that the `package_path` is the one getting stapled with the ticket so you shouldn’t need the zip 🤔 

https://github.com/fastlane/fastlane/blob/master/fastlane/lib/fastlane/actions/notarize.rb#L111

Reply:
Yea I don't know why it happens. I can give it another try now and tell you the exact error I'm seeing.

Reply:
Well, it worked this time 😓 haha. Thanks for making me check! 
I zipped up the file for distribution after notarizing and uploaded it. I don't know if it's because I used ditto to zip up like notarize does instead of zip. But it's working now. I'm gonna go ahead and close this PR. Thanks for your time!